recipes/libfabric/meta.yaml

```diff
@@ -0,0 +1,52 @@
+{% set version = "1.22.0" %}
+{% set build = 0 %}
+
+package:
+  name: libfabric
+  version: {{ version }}
+
+source:
+  url: https://github.com/ofiwg/libfabric/releases/download/v{{ version }}/libfabric-{{ version }}.tar.bz2
+  sha256: 485e6cafa66c9e4f6aa688d2c9526e274c47fda3a783cf1dd8f7c69a07e2d5fe
+
+build:
+  number: {{ build }}
+  skip: true  # [win]
+
+  run_exports:
+    - {{ pin_subpackage('libfabric', max_pin='x.y') }}
+  ignore_run_exports:
+    - libnuma
+    - libuuid
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ stdlib("c") }}
+    - libtool  # [unix]
+    - make  # [unix]
+  host:
+    - curl
```

Comment:
Configure needs both the curl headers and library

Reply:
`libcurl` [has the libraries and the headers](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Flinux-64%2Flibcurl-8.10.1-hbbe4b11_0.conda#files), and is the dependency to use in `host` if linking against curl. The `curl` package is the executable, and I think should ~never be in host (maybe `build` if a `curl` executable is needed.

But libcurl is not linked in the result, suggesting that it's not actually being used. The configure output referencing curl is only in `cxi` as far as I can see, which is ultimately not enabled, so while configure does find it, it does not appear to be using it to build anything. If curl is actually needed, I think there should probably be a test asserting `cxi` is present.

Reply:
With `curl` the configure does:
```
2024-10-27T13:33:08.3646116Z checking curl/curl.h usability... yes
2024-10-27T13:33:08.4037547Z checking curl/curl.h presence... yes
2024-10-27T13:33:08.4038445Z checking for curl/curl.h... yes
2024-10-27T13:33:08.4061948Z configure: looking for library without search path
2024-10-27T13:33:08.4745830Z checking for curl_global_init in -lcurl... yes
```
but when I tried with `libcurl` instead it did not find the headers

Reply:
OK, that is strange indeed. But if linking curl, `libcurl` must be present, whether `curl` is as well or not. But in any case, if we are not building `cxi`, I do not believe curl is needed.

I think you might be able to save some time configuring/discovering unavailable features if you explicitly `--enable-cxi=no` and other features you expect to be unavailable (opx, etc.). We might even get better, earlier errors if we set `--enable-usnic=yes` rather than the default `auto` for the optional features we do intend to be available.

Reply:
OK

Reply:
remove unused curl dependency:

```suggestion
```